src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Threading;
+using Avalonia.Controls;
+using Avalonia.Controls.ApplicationLifetimes;
+
+namespace Avalonia.Controls.ApplicationLifetimes
+{
+    public class ClassicDesktopStyleApplicationLifetime : IClassicDesktopStyleApplicationLifetime
+    {
+        private readonly Application _app;
+        private int _exitCode;
+        private CancellationTokenSource _cts;
+        private bool _isShuttingDown;
+
+        public ClassicDesktopStyleApplicationLifetime(Application app)
+        {
+            _app = app;
+            app.Windows.OnWindowClosed += HandleWindowClosed;
+        }
+        
+        /// <inheritdoc/>
+        public event EventHandler<ControlledApplicationLifetimeStartupEventArgs> Startup;
```

Comment:
How to I subscribe to this event before it is being raised? This should be possible within the AppBuilder. A Startup, Exit action would be enough I guess. Maybe I have missed something here.

Reply:
I guess we could move it to an earlier point of application initialization, so even platform initialization callbacks would have access to it via `AppBuilder.Instance.ApplicationLifetime`.

Reply:
Well, actually, it's already set before calling `SetupWithoutStarting()` here: https://github.com/AvaloniaUI/Avalonia/pull/2676/files#diff-01bc1e09e99bb7c8853294c6b0b206a2R94